File: backend/models/system_readiness.py
from __future__ import annotations

from datetime import datetime
from typing import Any, Literal

from pydantic import BaseModel, Field


CheckStatus = Literal["pass", "warn", "fail"]
OverallStatus = Literal["ready", "degraded", "blocked"]


class ReadinessCheck(BaseModel):
    id: str
    label: str
    status: CheckStatus
    required: bool
    message: str
    details: dict[str, Any] = Field(default_factory=dict)


class ReadinessApp(BaseModel):
    name: str = "VideoForge"
    version: str = "0.1.0"
    productionMode: bool = False


class ReadinessCapabilities(BaseModel):
    projectEditing: bool
    voiceover: bool
    previewRendering: bool
    jianyingZipExport: bool
    jianyingDirectExport: bool


class ReadinessSummary(BaseModel):
    passed: int
    warnings: int
    failed: int


class ReadinessResponse(BaseModel):
    status: OverallStatus
    checkedAt: datetime
    app: ReadinessApp
    checks: list[ReadinessCheck]
    capabilities: ReadinessCapabilities
    summary: ReadinessSummary

File: frontend/src/types/systemReadiness.ts
export type ReadinessStatus = 'ready' | 'degraded' | 'blocked'
export type ReadinessCheckStatus = 'pass' | 'warn' | 'fail'

export interface ReadinessCheck {
  id: string
  label: string
  status: ReadinessCheckStatus
  required: boolean
  message: string
  details: Record<string, unknown>
}

export interface ReadinessCapabilities {
  projectEditing: boolean
  voiceover: boolean
  previewRendering: boolean
  jianyingZipExport: boolean
  jianyingDirectExport: boolean
}

export interface SystemReadinessResponse {
  status: ReadinessStatus
  checkedAt: string
  app: { name: string; version: string; productionMode: boolean }
  checks: ReadinessCheck[]
  capabilities: ReadinessCapabilities
  summary: { passed: number; warnings: number; failed: number }
}
